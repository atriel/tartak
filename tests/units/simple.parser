number: "-"? digit
assignment: name assignment-oper number
