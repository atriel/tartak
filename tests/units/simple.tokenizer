regex: digit: "[0-9]+"

regex: name: "[a-zA-Z_]+"

string: assignment-oper: "="

# eos -- end of statement
string: eos: ";"
