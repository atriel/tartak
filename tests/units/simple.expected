{
    "tokens": [
        [0, 0, "answer"],
        [0, 0, "="],
        [0, 0, "42"],
        [0, 0, ";"]
    ]
}
